docs/runtime/types/module-struct.md

```diff
@@ -0,0 +1,200 @@
+
+### Implementations for Event
+
+Remember the `deposit_event` function defined in the `decl_module!` macro? It will call the `deposit_event` function in `srml-system` runtime module to deposit an event into block's event record.
+
+```rust
+impl<T: Trait> Module<T> {
+    fn deposit_event(event: Event<T>) {
+        <system::Module<T>>::deposit_event(<T as Trait>::from(event).into());
+    }
+}
+```
+
+Refer to [Event Enum](runtime/types/event-enum.md) for more knowledge around events in Substrate.
+
+### Utilities for Block Execution
+
+There are three reserved functions that can be defined in runtime modules:
+
+1. `on_initialize` function lets you express what should happen for your module when the block is beginning (right before the first extrinsic is applied).
+2. `on_finalize` function lets you express what should happen for your module when the block is ending.
+3. `offchain_worker` function allows you to perform long-running tasks that make validators generate extrinsics with the results of those long-running computations.
+
+The `decl_module!` macro will generate default implementations if you didn't provide your own:
+
+```rust
+impl<T: Trait> ::srml_support::runtime_primitives::traits::OnInitialize<T::BlockNumber>
+    for Module<T>
+{}
+
+impl<T: Trait> ::srml_support::runtime_primitives::traits::OnFinalize<T::BlockNumber>
+    for Module<T>
+{}
+
+impl<T: Trait> ::srml_support::runtime_primitives::traits::OffchainWorker<T::BlockNumber>
+    for Module<T>
+{}
+```
+
+## Tuple of All The Modules
+
+Your `Module` struct then comes along with other used SRML modules to form a tuple called `AllModules` which will be passed to the Executive module, handling dispatch to the various modules.
+
+```rust
+pub type System = system::Module<Runtime>;
+pub type Timestamp = timestamp::Module<Runtime>;
+pub type Consensus = consensus::Module<Runtime>;
+pub type Aura = aura::Module<Runtime>;
+pub type Indices = indices::Module<Runtime>;
+pub type Balances = balances::Module<Runtime>;
+pub type Sudo = sudo::Module<Runtime>;
+pub type TemplateModule = template::Module<Runtime>;
+
+type AllModules = (
+    Timestamp,
+    Consensus,
+    Aura,
+    Indices,
+    Balances,
+    Sudo,
+    TemplateModule,
+);
+
+/// Executive: handles dispatch to the various modules.
+pub type Executive = executive::Executive<Runtime, Block, Context, Balances, AllModules>;
+```
+
+`Executive` struct provides implementation to handle block execution, our tuple will be used during applying extrinsics. Let me stop here and leave remaining mystery for you to explore.
+
+## Metadata of Your Module
+
+`Module` struct provides query functions for metadata of module's dispatchable functions and storage, the metadata for `Event` enum is actually in its own implementation. The following json shows the metadata of our template runtime module. Go to [this page](TODO) for more details around Substrate runtime metadata.
+
+```json
```

Comment:
The Metadata is not Json. We should make this clear.

Reply:
I changed the expression to emphasize what we show is in JSON format. But still don't touch metadata details. The details will be put in another [page](https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/pull/163)